src/middleware/persist.ts

```diff
@@ -312,14 +312,14 @@ declare module '../vanilla' {
   }
 }
 
-type Write<T extends object, U extends object> = Omit<T, keyof U> & U
+type Write<T, U> = Omit<T, keyof U> & U
 type Cast<T, U> = T extends U ? T : U
 
 type WithPersist<S, A> = S extends { getState: () => infer T }
-  ? Write<S, StorePersist<Cast<T, object>, A>>
+  ? Write<S, StorePersist<Cast<T, unknown>, A>>
```

Comment:
@devanshj Does it make sense to keep `Cast`? Did it even make sense previously?

Reply:
> Did it even make sense previously?

Ah, it didn't pass because of `extends object` constraint.

Reply:
okay, I think it's fine to drop Cast entirely.

Reply:
Yes, we casted `T` to `object` so that we can pass it to `StorePersist` which required an `object`. So now that it doesn't require `object` we can simply drop the casting and just pass `T`.

Reply:
Good. I feel like it's more maintainable now.